src/System.Net.Primitives/tests/FunctionalTests/IPAddressParsingSpan.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Sockets;
+using Xunit;
+
+namespace System.Net.Primitives.Functional.Tests
+{
+    public sealed class IPAddressParsingSpan : IPAddressParsing
+    {
+        [Theory]
+        [MemberData(nameof(ValidIpv4Addresses))]
+        public void ParseIPv4Span_ValidAddress_Success(string address, string expected)
+        {
+            var addressArray = address.ToCharArray();
+            IPAddress ip = IPAddress.Parse(new ReadOnlySpan<char>(addressArray));
+
+            // Validate the ToString of the parsed address matches the expected value
+            Assert.Equal(expected, ip.ToString());
+            Assert.Equal(AddressFamily.InterNetwork, ip.AddressFamily);
+
+            // Validate the ToString representation can be parsed as well back into the same IP
+            addressArray = ip.ToString().ToCharArray();
+            IPAddress ip2 = IPAddress.Parse(new ReadOnlySpan<char>(addressArray));
+            Assert.Equal(ip, ip2);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidIpv4Addresses))]
+        public void TryParseIPv4Span_ValidAddress_Success(string address, string expected)
+        {
+            var addressArray = address.ToCharArray();
+            Assert.Equal(true, IPAddress.TryParse(new ReadOnlySpan<char>(addressArray), out IPAddress ip));
+
+            // Validate the ToString of the parsed address matches the expected value
+            Assert.Equal(expected, ip.ToString());
+            Assert.Equal(AddressFamily.InterNetwork, ip.AddressFamily);
+
+            // Validate the ToString representation can be parsed as well back into the same IP
+            addressArray = ip.ToString().ToCharArray();
+            Assert.Equal(true, IPAddress.TryParse(new ReadOnlySpan<char>(addressArray), out IPAddress ip2));
+            Assert.Equal(ip, ip2);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidIpv4Addresses))]
+        public void TryFormatIPv4_ValidAddress_Success(string address, string expected)
+        {
+            var result = new char[15];
+            var ip = IPAddress.Parse(address);
+
+            Assert.True(ip.TryFormat(new Span<char>(result), out int charsWritten));
+            Assert.Equal(expected.Length, charsWritten);
+            Assert.Equal(expected, new string(result, 0, charsWritten));
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidIpv4Addresses))]
+        public void TryFormatIPv4_ValidAddress_Failure(string address, string expected)
```

Comment:
Assuming this is the case, my suggestion would be to combine the success and failure tests for this, and validate three things:
- That it succeeds with exactly the right amount of output space
- That it fails with one less byte available than is needed, and none of the output is written
- That it succeeds with one more byte available than is needed, and that extra byte remains unwritten